superset/sql/parse.py

```diff
@@ -1503,3 +1503,31 @@ def sanitize_clause(clause: str, engine: str) -> str:
         )
     except SupersetParseError as ex:
         raise QueryClauseValidationException(f"Invalid SQL clause: {clause}") from ex
+
+
+def transpile_to_dialect(sql: str, target_engine: str) -> str:
+    """
+    Transpile SQL to the target database dialect using SQLGlot.
+
+    If the target engine is not in SQLGLOT_DIALECTS, returns the SQL as-is.
+    """
+    target_dialect = SQLGLOT_DIALECTS.get(target_engine)
+
+    # If no dialect mapping exists, return as-is
+    if target_dialect is None:
+        return sql
+
+    try:
+        parsed = sqlglot.parse_one(sql, dialect="postgres")
```

Comment:
Let's parse using the generic dialect here, since the use of "postgres" could be confusing:

```suggestion
        # use the generic parser
        parsed = sqlglot.parse_one(sql, sqlglot.dialects.dialect.Dialect)
```